### PR TITLE
Created docs for cudart function in torch.cuda

### LINKED ISSUE #127908: Document the torch.cuda.cudart function
Document the `torch.cuda.cudart` function located in [https://github.com/pytorch/pytorch/blob/main/torch/cuda/__init__.py#L318](https://github.com/pytorch/pytorch/blob/main/torch/cuda/__init__.py#L318).
Follow the [Docstring Guidelines](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines).

cc @brycebortree

### PR DESCRIPTION
**Pull request by ignaciobartol**
Fixes #127908

## Description

Created docs to document the torch.cuda.cudart function to solve the issue #127908. 
I tried to stick to the [guidelines to document a function](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#documenting-a-function) but I was not sure if there is a consensus on how to handle the docs of a function that calls an internal function. So I went ahead and tried what the function will raise, etc. from the user endpoint and documented it (i.e. I am giving what actually _lazy_init() will raise).  

### Summary of Changes

- Added docs for torch.cuda.cudart
- Added the cudart function in the autosummary of docs/source/cuda.rst

## Checklist 
- [X] The issue that is being fixed is referred in the description
- [X] Only one issue is addressed in this pull request
- [X] Labels from the issue that this PR is fixing are added to this pull request
- [X] No unnecesary issues are included into this pull request

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @svekars @brycebortree @ptrblck @msaroufim @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @LucasLLC @MeetVadakkanchery @mhorowitz

### CODE REVIEW

File: torch/cuda/__init__.py [hunk 1]
```diff
@@ -334,6 +334,29 @@ def _lazy_init():
...
+
+    Returns:
+        module: The CUDA runtime API module (_cudart).
+
+    Raises:
+        RuntimeError: If CUDA cannot be re-initialized in a forked subprocess.
+        AssertionError: If PyTorch is not compiled with CUDA support or if libcudart functions are unavailable.
+
+    Example:
+        >>> import torch
```
Comment (msaroufim) [inline torch/cuda/__init__.py:355]: So the usage instructions should do something useful and not sure what instantiating cudard() really does for most end users? So could you be more specific on examples of useful or common questions to ask

Comment (ignaciobartol) [reply torch/cuda/__init__.py:355]: Hey! Thanks for the comment. Based on the usage of the function [here](https://github.com/pytorch/pytorch/blob/c0b40ab42e38a208351911496b7153511304f8da/torch/cuda/profiler.py#L6) and in this [issue](https://github.com/pytorch/pytorch/issues/44206). What do you think about something along the lines provided below? I hope it makes sense, and I apologize if not; I am not very familiar with their usage as CUDA profiler, which seems to be the case for this function. If it still does not make sense to you, please provide me with some context where you would like to see an end-user example of this.  Thanks!


```python
import torch
from torch.cuda import cudart, check_error
import ctypes
import numpy as np

# Some random function to perform CUDA operations using streams
def perform_cuda_operations_with_streams():
    stream = torch.cuda.Stream()
    with torch.cuda.stream(stream):
        x = torch.randn(100, 100, device='cuda')
        y = torch.randn(100, 100, device='cuda')
        z = x @ y
    return z

def main():
    rt = cudart()

    # Start CUDA profiler
    check_error(rt.cudaProfilerStart())

    # Perform CUDA operations
    result = perform_cuda_operations_with_streams()
    print("CUDA operations completed.")

    # Stop CUDA profiler
    check_error(rt.cudaProfilerStop())

    # Create and destroy a CUDA stream
    stream = ctypes.c_ulonglong(0)
    stream_p = ctypes.POINTER(ctypes.c_void_p)(stream)
    stream_p_int = ctypes.cast(stream_p, ctypes.c_void_p).value
    check_error(rt.cudaStreamCreate(stream_p_int))
    print("CUDA stream created.")

    check_error(rt.cudaStreamDestroy(stream.value))
    print("CUDA stream destroyed.")

if __name__ == "__main__":
    main()

Comment (msaroufim) [reply torch/cuda/__init__.py:355]: this snippet does seem to instantiate cudart https://github.com/pytorch/pytorch/issues/44206

This part makes a bit more sense

```
    rt = cudart()

    # Start CUDA profiler
    check_error(rt.cudaProfilerStart())

    # Perform CUDA operations
    result = perform_cuda_operations_with_streams()
    print("CUDA operations completed.")

    # Stop CUDA profiler
    check_error(rt.cudaProfilerStop())
```

So great once you've done this where do you access the profile data? that could be a reasonable snippet

Comment (ignaciobartol) [reply torch/cuda/__init__.py:355]: Sorry I made a mistake in the commit @msaroufim, I am closing this PR and creating a new one with the updated example.

**Pull request closed without merging**